stl/inc/execution

```diff
@@ -5043,6 +5044,180 @@ _FwdIt2 adjacent_difference(_ExPo&&, const _FwdIt1 _First, const _FwdIt1 _Last,
+        }
+
+        _Uninitialized_value_construct_ivdep(_UFirst, _ULast);
+    } else if constexpr (remove_reference_t<_ExPo>::_Ivdep) {
+        _Uninitialized_value_construct_ivdep(_UFirst, _ULast);
+    } else {
+        if constexpr (_Use_memset_value_construct_v<_UnwrappedNoThrowFwdIt>) {
+            _Zero_range(_UFirst, _ULast);
+        } else {
+            for (; _First != _Last; ++_First) {
+                _Construct_in_place(*_UFirst);
+            }
+        }
+    }
+}
+
+template <class _UnwrappedNoThrowFwdIt, class _Diff>
+_UnwrappedNoThrowFwdIt _Uninitialized_value_construct_n_ivdep(_UnwrappedNoThrowFwdIt _UFirst, _Diff _Count) {
+    // value-initialize all elements in [_First, _First + _Count) assuming independent loop bodies
+    if constexpr (_Use_memset_value_construct_v<_UnwrappedNoThrowFwdIt>) {
+        _Zero_range(_UFirst, _UFirst + _Count);
+    } else {
+#pragma loop(ivdep)
+        for (; _Count > 0; --_Count, (void) ++_UFirst) {
+            _Construct_in_place(*_UFirst);
+        }
+    }
+
+    return _UFirst;
+}
+
+_EXPORT_STD template <class _ExPo, class _NoThrowFwdIt, class _Diff, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
+_NoThrowFwdIt uninitialized_value_construct_n(_ExPo&&, _NoThrowFwdIt _First, const _Diff _Count_raw) noexcept
+/* terminates */ {
+    // value-initialize all elements in [_First, _First + _Count_raw)
+    _REQUIRE_CPP17_MUTABLE_ITERATOR(_NoThrowFwdIt);
+    _Algorithm_int_t<_Diff> _Count = _Count_raw;
+
+    using _UnwrappedNoThrowFwdIt = _Unwrapped_t<const _NoThrowFwdIt&>;
+    auto _UFirst                 = _Get_unwrapped_n(_First, _Count);
+    auto _Ctor_fn                = [](decltype(*_UFirst) _Obj) { _Construct_in_place(_Obj); };
+    if (_Count > 0) {
+        if constexpr (remove_reference_t<_ExPo>::_Parallelize) {
+            const size_t _Hw_threads = __std_parallel_algorithms_hw_threads();
+            if (_Hw_threads > 1 && _Count >= 2) { // parallelize on multiprocessor machines with at least 2 elements
+                _TRY_BEGIN
+                _Static_partitioned_for_each2<decltype(_UFirst), decltype(_Count), decltype(_Ctor_fn)> _Operation{
+                    _Hw_threads, _Count, _Ctor_fn};
+                _Seek_wrapped(_First, _Operation._Basis._Populate(_Operation._Team, _UFirst));
+                _Run_chunked_parallel_work(_Hw_threads, _Operation);
+                return _First;
+                _CATCH(const _Parallelism_resources_exhausted&)
+                // fall through to serial case below
+                _CATCH_END
+            }
+
+            _Seek_wrapped(_First, _Uninitialized_value_construct_n_ivdep(_UFirst, _Count));
+        } else if constexpr (remove_reference_t<_ExPo>::_Ivdep) {
+            _Seek_wrapped(_First, _Uninitialized_value_construct_n_ivdep(_UFirst, _Count));
+        } else {
+            if constexpr (_Use_memset_value_construct_v<_UnwrappedNoThrowFwdIt>) {
+                _Zero_range(_UFirst, _UFirst + _Count);

```

Comment:
You appear to be discarding the return value of `_Zero_range` here, however below you use `_Seek_wrapped` to seek first to the "new" first value before returning it. As far as I can tell this means you are returning an incorrect value in the case where this branch is hit.